### ISSUE TITLE
New Rule: no-heading-like-paragraph

### ISSUE THREAD

**Issue by Gaic4o** [eslint/markdown#700]
### Rule details

This rule should report when the first line of a paragraph starts with seven or more hash characters (`#`) followed by a space or tab.  CommonMark ATX headings support one to six hash characters. Therefore, content such as `####### Installation` is parsed as a regular paragraph rather than a heading.

### What type of rule is this?

Warns about a potential problem

### Example code

```markdown
// The following code should trigger a warning
####### Installation

######## Configuration
```

### Prior Art

remark-lint already provides a rule with the same purpose:

[`remark-lint-no-heading-like-paragraph`](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-no-heading-like-paragraph)

This rule also warns about paragraphs that start with seven or more hash characters and appear to be intended as headings.

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

 Disclosure: I'm a participant of [open source contribution program OSSCA](https://github.com/eslint-ossca)

**Comment by DMartens** [comment]
I am unsure how [applicable](https://eslint.org/docs/latest/contribute/propose-new-rule#core-rule-guidelines) this rule is as I never saw a case for this in the wild.
Searching for the pattern in [public repositories](https://sourcegraph.com/search?q=context%3Aglobal+%5E%23%23%23%23%23%23%23+&patternType=regexp&sm=0&df=%5B%22lang%22%2C%22Markdown%22%2C%22lang%3Amarkdown%22%5D&__cc=1) I could not find a match.
Would like another team member opinion on this.

**Comment by lumirlumir** [comment]
I’m personally a soft +1 on this suggestion.

Since we’ve generally accepted rule addition requests from `markdownlint` and `remark-lint` that aren’t about stylistic choices in this repo, I think this rule could be a nice addition.

For example, I had a similar opinion on https://github.com/eslint/markdown/issues/419#issuecomment-2975550723 regarding how useful the rule would be, but it was ultimately accepted.

**Comment by lumirlumir** [comment]
Could I get a few more opinions on this? @eslint/eslint-team

**Comment by nzakas** [comment]
In the interest of Remark-lint compat, I'm in favor. 👍
